docs/rfc/rfc-019-block-structure-consolidation.md

```diff
@@ -0,0 +1,363 @@
+The `PartSetHeader` is used by nodes to gossip the block by dividing it into
+parts. Nodes receive the `PartSetHeader` from their peers, informing them of
+what pieces of the block to gather. There is no strong reason to include this
+value in the block. Validators will still be able to gossip and validate the
+blocks that they received from their peers using this mechanism even if it is
+not written into the block. The `BlockID` can therefore be consolidated into
+just the hash of the block. This is by far the most uncontroversial change
+and there appears to be no good reason _not_ to do it. Further evidence that
+the field is not meaningful can be found in the fact that the field is not
+actually validated to ensure it is correct during block validation. Validation
+only checks that the [field is well formed][psh-check].
+
+#### ChainID
+
+The `ChainID` is a string selected by the chain operators, usually a
+human-readable name for the network. This value is immutable for the lifetime
+of the chain and is defined in the genesis file. It is therefore hashed into the
+original block and therefore transitively included as in the Merkle root hash of
+every block. The redundant field is a candidate for removal from the root hash
+of each block. However, aesthetically, it's somewhat nice to include in each
+block, as if the block was 'stamped' with the ID. Additionally, re-validating
+the value from genesis would be painful and require reconstituting potentially
+large chains. I'm therefore mildly in favor of maintaining this redundant
+piece of information. We pay almost no storage cost for maintaining this
+identical data, so the only cost is in the time required to hash it into the
+structure.
+
+#### LastResultsHash
+
+`LastResultsHash` is a hash covering the result of executing the transactions
+from the previous block. It covers the response `Code`, `Data`, `GasWanted`,
+and `GasUsed` with the aim of ensuring that execution of all of the transactions
+was performed identically on each node. The data covered by this field _should_
+be also reflected in the `AppHash`. The `AppHash` is provided by the application
+and should be deterministically calculated by each node. This field could
+therefore be removed on the grounds that its data is already reflected elsewhere.
+
+I would advocated for keeping this field. This field provides an additional check
+for determinism across nodes. Logic to update the application hash is more
+complicated for developers to implement because it relies either on building a
+complete view of the state of the application data. The `Results` returned by
+the application contain simple response codes and deterministic data bytes.
+Leaving the field will allow for transaction execution issues that are not
+correctly reflected in the `AppHash` to be more completely diagnosed.
+
+Take the case of mismatch of `LastResultsHash` between two nodes, A and B, where both
```

Comment:
It seems like the more likely scenario is that A and B report _different_ AppHashes for the same cumulative transaction hash? Not that it couldn't be the opposite too, but if I saw that in a trace I'd assume the _node_ had computed the wrong values rather than the app diverging.